pkgs/development/libraries/libirecovery/default.nix

```
@@ -0,0 +1,51 @@
+{ stdenv, fetchFromGitHub, automake, autoconf, libtool, pkgconfig
+, libusb
+, readline
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libirecovery";
+  version = "2018-01-28";
+
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "libimobiledevice";
+    repo = pname;
+    rev = "5da2a0d7d60f79d93c283964888c6fbbc17be1a3";
+    sha256 = "0fqmr1h4b3qn608dn606y7aqv3bsm949gx72b5d6433xlw9b23n8";
+  };
+
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    libtool
+    pkgconfig
+  ];
+  propagatedBuildInputs = [
+    libusb
+    readline
+  ];
+
+  preConfigure = "NOCONFIGURE=1 ./autogen.sh";
+
+  configureFlags = [
+    "--with-udevrulesdir=$out/lib/udev/rules.d"
```

```suggestion
    "--with-udevrulesdir=${placeholder ''out''}/lib/udev/rules.d"
```

Also, does this need a nixos configuration?

---

That looks sensible, is `placeholder` explained somewhere? I haven't seen it yet.

> Also, does this need a nixos configuration?

I am not sure, I've only used the `idevicerestore` binary so far, and do not know what other downstreams of the lib can do if more is done with the udev rule. I'll add a comment so that other's doing more with it have an easier time.

(CC @matthewbauer because https://aur.archlinux.org/packages/libirecovery-git/ suggests you know at least a little bit about this package, just in case.)

---

> That looks sensible, is placeholder explained somewhere? I haven't seen it yet.

> Derivation attributes can now reference the outputs of the derivation using the `placeholder` builtin function.

I couldn't find anymore than that. So the source is probably the docs.

> I am not sure, I've only used the idevicerestore binary so far, and do not know what other downstreams of the lib can do if more is done with the udev rule. I'll add a comment so that other's doing more with it have an easier time.

It seems they exist to allow non-root users access to DFU/Recovery mode https://github.com/libimobiledevice/libirecovery/commit/b1bc747824c21749d3d09c1aa6b1ce71087cd8cd.
Seems kinda particular.